data-tck/src/main/java/io/micronaut/data/tck/repositories/SaleRepository.java

```diff
@@ -38,4 +41,15 @@ public interface SaleRepository extends CrudRepository<Sale, Long> {
 
     void updateData(@Id Long id, @Parameter("data") Map<String, String> data, @Parameter("dataList") List<String> dataList);
 
+    @Query("SELECT extra_data AS extraData FROM sale WHERE id = :id")
+    @QueryResult(column = "extraData", queryResultType = QueryResult.QueryResultType.JSON)
+    Optional<Discount> getDiscountById(Long id);
+
+    @Query("SELECT extra_data AS DATA FROM sale WHERE name = :name")
+    @QueryResult(column = "DATA", queryResultType = QueryResult.QueryResultType.JSON)
+    List<Sale> findAllByNameFromJson(String name);
+
+    @Query("SELECT extra_data AS DATA FROM sale WHERE name = :name")
+    @QueryResult(column = "DATA", queryResultType = QueryResult.QueryResultType.JSON)
+    Optional<Sale> findByNameFromJson(String name);
```

Comment:
Can you add one test with the type tabular?

Reply:
Addded